cmd/tests/events/events.go

```diff
@@ -0,0 +1,74 @@
+// Package events is used for testing the event functionality.
+package events
+
+import (
+	"fmt"
+	"sync"
+
+	"go.k6.io/k6/api/v1/client"
+	"go.k6.io/k6/event"
+	"go.k6.io/k6/js/modules"
+)
+
+// RootModule is the global module instance that will create module
+// instances for each VU.
+type RootModule struct {
+	initOnce        sync.Once
+	apiAddress      string
+	subscribeEvents []event.Type
+}
+
+// Events represents an instance of the events module.
+type Events struct{}
+
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &Events{}
+)
+
+// New returns a pointer to a new RootModule instance.
+func New(apiAddress string, subscribeEvents []event.Type) *RootModule {
+	return &RootModule{
+		initOnce:        sync.Once{},
+		apiAddress:      apiAddress,
+		subscribeEvents: subscribeEvents,
+	}
+}
+
+// NewModuleInstance implements the modules.Module interface to return
+// a new instance for each VU.
+func (rm *RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	rm.initOnce.Do(func() {
+		sid, evtCh := vu.Events().Subscribe(rm.subscribeEvents...)
+		logger := vu.InitEnv().Logger
+		go func() {
+			api, _ := client.New(rm.apiAddress)
+			for {
+				select {
+				case evt, ok := <-evtCh:
+					if !ok {
+						return
+					}
+					var testStatus string
+					if evt.Type != event.Exit {
+						status, _ := api.Status(vu.Context())
+						testStatus = fmt.Sprintf(", test status: %s", status.Status.String())
+					}
+					logger.Infof("got event %s with data '%+v'%s", evt.Type, evt.Data, testStatus)
```

Comment:
This is a test module, and not used by k6 users. This is logged as `Info` since this output is read by the test. I suppose it could be lower severity, but we'd need to change [the `LoggerHook` level in `NewGlobalTestState()`](https://github.com/grafana/k6/blob/8f6f73b17f441054090f6fb8957b94d8ef551bad/cmd/tests/test_state.go#L50) as well, and practically there wouldn't be a difference. So I'd rather keep it as is, if you don't mind.